### PR TITLE
Im gui

### PR DESCRIPTION
**Pull request by BeardyKing**

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -13,6 +13,7 @@
 [submodule "third/imgui"]
 	path = third/imgui
 	url = https://github.com/ocornut/imgui.git
+	branch = docking
```
Comment (lmariscal) [inline .gitmodules:16]: https://git-scm.com/docs/gitmodules

Didn't know this was possible :ok_hand: .

**Review by lmariscal — CHANGES_REQUESTED** [review]
Minor details, overall great.

File: core/src/engine.cpp [hunk 2]
```diff
@@ -48,9 +49,11 @@ void Engine::update_modules() {
     for (auto& module : m_engineModules) {
         module->on_late_update();
     }
+    bgfx::frame();
```
Comment (BertTDev) [inline core/src/engine.cpp:52]: Should this be handled inside one of the renderers?

**Review by BertTDev — COMMENTED** [review]
Very nice, look forward to putting this altogether in the editor

File: core/src/engine.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (BeardyKing) [reply core/src/engine.cpp:52]: I agree this bgfx function should be moved to some internal engine function such as 'swap_frame()' as that's what it's doing

File: core/src/transform.cpp [hunk 3]
```diff
@@ -42,9 +42,9 @@ void Transform::set_rotation_euler(const vec3& euler) {
     m_rotation = quat(radians(euler));
 }
 
-mat4 Transform::get_model_matrix() const {
-    mat4 modelMatrix{1.0f};
-    modelMatrix = translate(mat4(1.0f), m_position) * toMat4(m_rotation) * scale(mat4(1.0f), m_scale);
+glm::mat4 Transform::get_model_matrix() const {
```
[the remark this answers is not in the document]
Comment (BeardyKing) [reply core/src/transform.cpp:45]: oops this was added for the screenshot for the technical document and wasn't meant to be updated for the actual project

File: core/src/window.cpp [hunk 4]
```diff
@@ -22,7 +22,7 @@ namespace knot {
 
 Window::Window(int width, int height, std::string title, Engine& engine)
     : m_width(width), m_height(height), m_title(title), m_window(nullptr), m_engine(engine) {
-    int glfw_init_res = glfwInit();
+    int glfw_init_res = glfwInit(), debug_editor_resize_flag(true);
```
[the remark this answers is not in the document]
Comment (BeardyKing) [reply core/src/window.cpp:25]: this is meant to be inside of the initialization of the class and has been changed

File: core/inc/knoting/window.h [hunk 5]
```diff
@@ -33,13 +33,31 @@ class Window : public Subsystem {
 
     int get_window_width() { return m_width; };
     int get_window_height() { return m_height; };
+    void set_window_height(int height) { m_height = height; };
```
[the remark this answers is not in the document]
Comment (BeardyKing) [reply core/inc/knoting/window.h:36]: will remove set_window_width/height and replace with a 'set_window_size(vec2i size); function that calls 'recreate_framebuffer' with the new member variables that are set

**Review by BertTDev — COMMENTED** [review]
LGTM!

**Review by lmariscal — APPROVED** [review]
LGTM

**Pull request merged**